### PR TITLE
SGT: add batchDepositForAll

### LINKED ISSUE #120: Need a method to deposit SGT all with the same balance
This will save the calldata size by half (although some zeros will be optimized during batch compression)

### PR DESCRIPTION
**Pull request by blockchaindevsh**
This PR fixes https://github.com/ethstorage/optimism/issues/120

### CODE REVIEW

**Review by qizhou — APPROVED** [review]

File: packages/contracts-bedrock/src/L2/SoulGasToken.sol [hunk 1]
```diff
@@ -80,6 +80,16 @@ contract SoulGasToken is ERC20Upgradeable, OwnableUpgradeable {
         require(msg.value == totalValue, "unexpected msg.value");
     }
 
+    /// @notice batchDepositForAll is similar to batchDepositFor, but the value is the same for all accounts.
+    function batchDepositForAll(address[] calldata accounts, uint256 value) external payable {
```
Comment (qzhodl) [inline packages/contracts-bedrock/src/L2/SoulGasToken.sol]: Do we want to follow the OP's coding style here?

Comment (blockchaindevsh) [reply packages/contracts-bedrock/src/L2/SoulGasToken.sol]: Good catch, fixed!

File: packages/contracts-bedrock/src/L2/SoulGasToken.sol [hunk 2]
```diff
@@ -80,6 +80,16 @@ contract SoulGasToken is ERC20Upgradeable, OwnableUpgradeable {
         require(msg.value == totalValue, "unexpected msg.value");
     }
 
+    /// @notice batchDepositForAll is similar to batchDepositFor, but the value is the same for all accounts.
+    function batchDepositForAll(address[] calldata _accounts, uint256 _value) external payable {
+        require(IS_BACKED_BY_NATIVE, "batchDepositForAll should only be called when IS_BACKED_BY_NATIVE");
```
Comment (qzhodl) [inline packages/contracts-bedrock/src/L2/SoulGasToken.sol]: According to the coding style, the assert need to begin with "SGT: "?

Comment (blockchaindevsh) [reply packages/contracts-bedrock/src/L2/SoulGasToken.sol]: Done, it's really easy to forget when surrounding examples are not using the style..

**Review by qzhodl — APPROVED** [review]

**Pull request merged**